src/porepy/models/constitutive_laws.py

```diff
@@ -1268,6 +1356,699 @@ def interface_vector_source(
+        # the change of meaning of subscript is unfortunate, but it is very important to
+        # understand how the components of the permeability tensor and the normal and
+        # distance vectors are multiplied.) This formulation can be reformulated to
+        #
+        #   t = n_r^T e_s K_rs / dist
+        #
+        # where the point is that, by right multiplying the permeability tensor, this
+        # can be represented as an Ad operator (which upon parsing will be an AdArray
+        # which only can be right multiplied). The below code implements this
+        # formulation. The full transmissibility matrix is obtained by taking the
+        # harmonic mean of the two half-face transmissibilities on each face.
+
+        # The cell-wise permeability tensor is represented as an Ad operator which
+        # evaluates to an AdArray with 9 * n_cells entries. Also scale with specific
+        # volume.
+        basis = self.basis(subdomains, dim=9)  # type: ignore[call-arg]
+        volumes = pp.ad.sum_operator_list(
+            [e @ self.specific_volume(subdomains) for e in basis]
+        )
+        k_c = volumes * diffusivity_tensor(subdomains)
+
+        # Create the helper discretization object, which will be used to generate
+        # grid-related quantities and mappings.
+        diff_discr = pp.numerics.fv.tpfa.DifferentiableTpfa()
+
+        # Get the normal vector, vector from cell center to face center (d_vec), and
+        # distance from cell center to face center (dist) for each half-face.
+        n, d_vec, dist = diff_discr.half_face_geometry_matrices(subdomains)
+
+        # Compose the geometric part of the half-face transmissibilities. Note that
+        # dividing d_vec by dist essentially forms a unit vector from cell to face
+        # center.
+        d_n_by_dist = sps.diags(1 / dist) * d_vec @ n
+
+        # Form the full half-face transmissibilities and take its reciprocal, preparing
+        # for a harmonic mean between the two half-face transmissibilities on ecah side
+        # of a face.
+        one = pp.ad.Scalar(1)
+        t_hf_inv = one / (pp.ad.SparseArray(d_n_by_dist) @ k_c)
+
+        # Compose full-face transmissibilities
+        # Sum over half-faces to get transmissibility on faces.
+        # Include sign to cancel the effect of the d_vec @ n having opposite signs on
+        # the two half-faces.
+        hf_to_f = diff_discr.half_face_map(
+            subdomains, to_entity="faces", with_sign=True
+        )
+        # Take the harmonic mean of the two half-face transmissibilities.
+        t_f_full = one / (pp.ad.SparseArray(hf_to_f) @ t_hf_inv)
+        t_f_full.set_name("transmissibility matrix")
+        return t_f_full, diff_discr, hf_to_f, d_vec
+
+    def __mpfa_flux_discretization(
+        self, base_discr: pp.ad.MpfaAd, T_f: ArrayType, p_diff: ArrayType, p: ArrayType
+    ) -> ArrayType:
+        """Take the differential of the flux associated with the potential difference.
```

Comment:
Here and in next method: Would it be helpful to point out that this is an approximation of the product rule for mpfa_transmissibility * p and spell out the terms?

Reply:
Yes, will do.

Reply:
Implemented.